project/api/models.py

```diff
@@ -88,10 +97,50 @@ def __str__(self):
 
 class DiplomaticRelation(models.Model):
     """
-    Abstract class for relationships between states, including wars, alliances, overlordships, etc.
-    This would be a whole 'nother project that should probably be primarily populated by Wikipedia scraping
-    Some relations will most likely be controversial, for example the earldoms of Ireland swore allegiance to
-        the English monarchs but in otherwise acted entirely independently, maybe have an autonomy slider
-        or something similar?
+    Defines political and diplomatic interactions between Nations.
     """
-    pass
+    start_date = models.DateField(help_text="When this relation takes effect")
+    end_date = models.DateField(help_text="When this relation ceases to exist")
+    first_party = models.ManyToManyField(Nation, related_name='first_parties')
+    second_party = models.ManyToManyField(Nation, related_name='second_parties')
```

Comment:
Apart from "at war", all diplomatic relations are hierarchical. (e.g. military alliance would be an entity having two child entities).

I'd be more in favor of calling it parent_party, child_party.

Reply:
Good idea, frankly I had no idea what to call this field

Reply:
This question made me think this model might want to be a ManyToMany through field (https://docs.djangoproject.com/en/2.1/topics/db/models/#extra-fields-on-many-to-many-relationships)

Reply:
@dwaxe interesting, wasn't aware of such a field. I'll test this out.

Reply:
Correct me if my understanding of your idea is wrong, but after some more research I don't think a m2m through field on the Nation model would be the best option. To my knowledge this setup would entail every parent Nation having a relationship with a group of children Nations through the DiplomaticRelation model. If we had multiple parents they would each have to set this field with the same value of child nations which would be redundant as opposed to having a collection of parent nations as one field and a collection of children as the second.